src/app/components/study-accordion/study-accordion.directive.js

```diff
@@ -43,25 +42,50 @@ angular.module('transmartBaseUi')
 
         node.isLoading = true;
         TreeNodeService.getNodeChildren(node).then(function (result) {
-          console.log('after getNodeChildren')
           node.isLoading = false;
         });
-
       };
 
+      $scope.prev_node = null;
+      $scope.clearMetadata = function (node) {
+        //reset or toggle the flags
+        var isSame = false;
+        if ($scope.prev_node !== null && $scope.prev_node.$$hashKey == node.$$hashKey) {
+          isSame = true;
+        }
+        if (!isSame && $scope.prev_node !== null) {
+          $scope.prev_node.isPopOpen = false;
+        }
+        node.isPopOpen = !node.isPopOpen;
+        $scope.prev_node = node;
+
+        //clear the html
+        $scope.metadataObj = {};
+        var query = document.getElementsByClassName("popover");
+        var elms = angular.element(query);
```

Comment:
can we make it clear that we are clearing all visible popover elements here. Maybe call it **popoverElements** instead of **elms**